### ISSUE TITLE
Provide additional context for "no output generated" messages

### ISSUE THREAD

**Issue by christophermaier** [operable/cog#959]
Currently when Cog finishes a pipeline stage and there is no output to pass to downstream stages, a message is output saying ["Pipeline executed successfully, but no output was returned"](https://github.com/operable/cog/blob/1229b19a22f217273ed6617c114d0055f3aff0ec/priv/templates/common/early-exit.greenbar). This is not terribly useful, and potentially confusing. The entire pipeline hasn't necessarily been executed; if the second stage of a 4-stage pipeline returns no output, we halt the pipeline early without executing the remaining stages.

We should indicate how far into the pipeline the execution got before halting. I think all the necessary information should be accessible in the executor at that point; it's just a matter of packaging it up for the template (and editing the template, naturally).

**Comment by utsysadmin** [comment]
Yes please!  I'm having this problem and have zero information as to why.  The logs on the relay show an invocation request, an environment setup, no dynamic config found, and then no data related to a failure at all.
